website/static/js/components/legend.js

```diff
@@ -2,16 +2,39 @@ require('css/legend.css');
 var m = require('mithril');
 
 module.exports = {
-    view: function(data, repr, opts) {
-        if(data[0].label){
-            data.sort(function(a, b) {
+    view: function (data, repr, opts) {
+        if (data[0].label) {
+            /* sorting should be:
+               [ (IN alpha order), Other, Uncategorized]
+
+                sort a b
+                    a/b == Uncat     = -1
+                    a/b == Other
+                        b/a == Uncat =  1
+                        otherswise   = -1
+                    otherwise        = compare a b
+             */
+            data.sort(function (a, b) {
+                // Uncategorized should come last
+                if(a.label === 'Uncategorized'|| b.label === 'Uncategorized') return -1;
```

Comment:
This will work, but only because `a.label` won't ever be `'Uncategorized'` with current initial ordering. 

Returning `-1` in the function that `sort` takes signifies the two items should not swap places, so if it was (somehow) `a.label`, `Uncategorized` wouldn't be last in the final result.